### PR TITLE
files: Don't use canonical path when serving file

### LINKED ISSUE #2128: actix-files::Files: symlinks are served with the filename of the original file.
## Expected Behavior

When serving a symbolic link with `actix-files::Files` service, the `content-disposition` header should be set to the filename of the symlink.

## Current Behavior

The filename of the original file is used.

## Possible Solution

Don't use the canonical path when serving a file?
https://github.com/actix/actix-web/blob/f66774e30b239546991fb6c0d54c8885171d1b8f/actix-files/src/service.rs#L86

## Context

Originally reported in svenstaro/miniserve#462

## Your Environment

* Rust Version (I.e, output of `rustc -V`): rustc 1.50.0
* Actix Web Version: 3
* actix-files version: 0.5

### PR DESCRIPTION
**Pull request by aliemjay**
## PR Type

Bug Fix

## PR Checklist

- [x] Tests for the changes have been added / updated.
- [x] A changelog entry has been made for the appropriate packages.
- [x] Format code with the latest stable rustfmt.
- [x] (Team) Label with affected crates and semver status.

## Overview

Otherwise, for symlink files, the content-disposition header would be set to the filename of the pointed-to file.

Closes #2128

### CODE REVIEW

**Comment by aliemjay** [comment]
Is it necessary to add a test case and a changelog entry for such change?

**Comment by ibraheemdev** [comment]
Adding a test case for the bug fix is a good idea.

**Comment by aliemjay** [comment]
Just added a test case. I don't know if a changelog entry is required?

**Comment by robjtede** [comment]
> I don't know if a changelog entry is required?

Yes please.

File: actix-files/src/service.rs [hunk 1]
```diff
@@ -83,10 +83,10 @@ impl Service<ServiceRequest> for FilesService {
...
 
         // full file path
-        let path = match self.directory.join(&real_path).canonicalize() {
-            Ok(path) => path,
-            Err(err) => return Box::pin(self.handle_err(err, req)),
-        };
+        let path = self.directory.join(&real_path);
+        if let Err(err) = path.canonicalize() {
+            return Box::pin(self.handle_err(err, req));
+        }
```
Comment (robjtede) [inline actix-files/src/service.rs:89]: My concern about this is that we probably want to normalize the path in the way that canonicalize does just without following symlinks. Have you done any testing with paths that includes '../' and such? I'm not sure we have test coverage for it.

Comment (aliemjay) [reply actix-files/src/service.rs:89]: I think `real_path` is already normalized by `PathBufWrap::parse_path()` and `self.directory` is also a canonical path (see `Files::new()`) , so I think this simple join should pose no problem.

I've just tested it with '../' in both url path and `self.directory`. If you wish I can modify the testcase for this?

Alternatively, we can keep the old code and use `NamedFile::from_file()` to pass the correct filename.

Comment (robjtede) [reply actix-files/src/service.rs:89]: It's okay then, I don't see any immediate problems with the change made.

**Review by robjtede — APPROVED** [review]

**Pull request merged**